tests/devices/qubit/test_jaxpr_adjoint_jvp.py

```diff
@@ -0,0 +1,246 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test for default.qubits execute_and_jvp method."""
+from functools import partial
+
+import pytest
+
+import pennylane as qml
+
+jax = pytest.importorskip("jax")
+jnp = pytest.importorskip("jax.numpy")
+
+from pennylane.devices.qubit.jaxpr_adjoint import (  # pylint: disable=wrong-import-position
+    execute_and_jvp,
+)
+
+pytestmark = [pytest.mark.jax, pytest.mark.usefixtures("enable_disable_plxpr")]
+
+
+class TestErrors:
+    """ "Test explicit errors for various unsupported cases."""
+
+    def test_no_differentiable_op_math(self):
+        """Test an error is raised if we have differentiable operator arithmetic."""
+
+        def f(x):
+            qml.adjoint(qml.RX(x, 0))
+            return qml.expval(qml.Z(0))
+
+        jaxpr = jax.make_jaxpr(f)(0.5)
+
+        with pytest.raises(NotImplementedError):
+            execute_and_jvp(jaxpr.jaxpr, (0.5,), (1.0,), num_wires=1)
+
+    def test_only_expvals(self):
+        """Test that an error is raised for other measurments."""
+
+        def f(x):
+            qml.RX(x, 0)
+            return qml.probs(wires=0)
+
+        jaxpr = jax.make_jaxpr(f)(0.5).jaxpr
+
+        with pytest.raises(NotImplementedError, match="expectations of observables"):
+            execute_and_jvp(jaxpr, (0.5,), (1.0,), num_wires=1)
+
+    def test_no_for_loop(self):
+        """Generic test for a primitive without a registered jvp rule."""
+
+        def f(x):
+            @qml.for_loop(3)
+            def g(i):
+                qml.RX(x, i)
+
+            g()
+            return qml.expval(qml.Z(0))
+
+        jaxpr = jax.make_jaxpr(f)(0.5).jaxpr
+
+        with pytest.raises(NotImplementedError, match="does not have a jvp rule."):
+            execute_and_jvp(jaxpr, (0.5,), (1.0,), num_wires=1)
+
+    def test_at_most_one_trainable_param(self):
+        """Test that multiple arguments are not trainable."""
+
+        def f(x, y, z):
+            qml.Rot(x, y, z, 0)
+            return qml.expval(qml.Z(0))
+
+        jaxpr = jax.make_jaxpr(f)(0.5, 0.6, 0.7).jaxpr
+
+        with pytest.raises(NotImplementedError, match="only differentiable parameters in the 0"):
+            execute_and_jvp(jaxpr, (1.0, 2.0, 3.0), (1.0, 1.0, 1.0), num_wires=1)
+
+    def test_capture_renabled_if_generator_failure(self):
+        """Test that capture stays enabled if the generator is undefined."""
+
+        def f(x):
+            qml.Rot(x, 1.2, 2.3, wires=0)
+            return qml.expval(qml.Z(0))
+
+        jaxpr = jax.make_jaxpr(f)(0.5).jaxpr
+
+        with pytest.raises(ValueError):
```

Comment:
Shouldn't this a generator-undefined related error? :thinking:

Reply:
That's what I thought.  I'm planning on fixing that at some point soon, but for now, `ValueError` is the behavior.